lib/methods/api-request.js

```diff
@@ -50,20 +60,19 @@ const apiRequest = client => {
               if (data && ((status === 401 && data.error_code === 102) || status === 412)) {
                 // 'Authentication failed for this user ID' or no store found
                 // remove current authentication
-                let sql = 'DELETE FROM ' + table + ' WHERE authentication_id = ?'
-                let params = [ myId ]
-
-                // run query
-                db.run(sql, params, err => {
-                  if (err) {
-                    // SQL error ?
-                    throw err
-                  } else {
-                    // mark returned error object
-                    error.appAuthRemoved = true
+                if(dbFilename){
+                  let sql = 'DELETE FROM ' + table + ' WHERE authentication_id = ?'
+                  let params = [ myId ]
+                  // run query
+                  db.run(sql, params, err => handleRejectPromisse(err, error, reject))
+                }else{
+                  try {
+                    db.collection(table).doc(myId).delete
```

Comment:
acho que isto é uma função, não?

```suggestion
                    db.collection(table).doc(myId).delete()
```

Reply:
e não seria melhor um `catch` na promise mesmo em vez de try aqui (já que você não tá usando async/await mesmo)?

```suggestion
                    db.collection(table).doc(myId).delete()
                      .catch(err => handleRejectPromisse(err, error, reject))
```